### PR TITLE
makes people with the claustrophobia quirk fear santa claus

### PR DESCRIPTION
**Pull request by Kylerace**
## About The Pull Request
https://youtu.be/IEKMotEN5Z4?t=11

## Why It's Good For The Game
makes the phobia true to its name

## Changelog

:cl:
add: People with the claustrophobia quirk will now be afraid of Santa Claus and certain Claus-like entities
/:cl:

### CODE REVIEW

**Comment by Bawhoppen** [comment]
SMASH THAT SPEEDMERGE BUTTON

**Comment by ArcaneMusic** [comment]
Put the RATIONAL in IRRATIONAL FEAR, DOC

**Comment by MrMelbert** [comment]
son of a bitch sees you when you're sleeping, and you're all okay with it?

**Comment by timothymtorres** [comment]
Christmas is canceled kiddos

File: code/datums/quirks/negative.dm [hunk 1]
```diff
@@ -867,11 +867,38 @@
...
+
+	for(var/mob/living/carbon/human/possible_claus in view(5, quirk_holder))
+		if(evaluate_jolly_levels(possible_claus))
+			nick_spotted = TRUE
+			break
+
+	if(!nick_spotted)
+		if(isturf(quirk_holder.loc))
+			SEND_SIGNAL(quirk_holder, COMSIG_CLEAR_MOOD_EVENT, "claustrophobia", /datum/mood_event/claustrophobia)
+			return
```
Comment (LemonInTheDark) [inline code/datums/quirks/negative.dm:880]: *dies*

File: code/datums/quirks/negative.dm [hunk 2]
```diff
@@ -867,11 +867,38 @@
...
 	quirk_holder.losebreath += 0.25 // miss a breath one in four times
 	if(DT_PROB(25, delta_time))
-		to_chat(quirk_holder, span_warning("You feel trapped!  Must escape... can't breath..."))
+		if(nick_spotted)
+			to_chat(quirk_holder, span_warning("Santa Claus is here! I gotta get out of here!"))
+		else
+			to_chat(quirk_holder, span_warning("You feel trapped!  Must escape... can't breath..."))
+
+///investigates whether possible_saint_nick possesses a high level of christmas cheer
+/datum/quirk/claustrophobia/proc/evaluate_jolly_levels(mob/living/carbon/human/possible_saint_nick)
```
Comment (LemonInTheDark) [inline code/datums/quirks/negative.dm:891]: ~~could this be inlined?~~

**Review by LemonInTheDark — APPROVED** [review]

File: code/datums/quirks/negative.dm [hunk 3]
```diff
@@ -867,11 +867,38 @@
...
-	
+	var/nick_spotted = FALSE
+
+	for(var/mob/living/carbon/human/possible_claus in view(5, quirk_holder))
+		if(evaluate_jolly_levels(possible_claus))
+			nick_spotted = TRUE
+			break
+
+	if(!nick_spotted)
+		if(isturf(quirk_holder.loc))
```
Comment (MrMelbert) [inline code/datums/quirks/negative.dm:878]:
```suggestion
	if(!nick_spotted && isturf(quirk_holder.loc))
```
?

File: code/datums/quirks/negative.dm (hunk 1 above)
Comment (tralezab) [reply code/datums/quirks/negative.dm:880]: lmfao

File: code/datums/quirks/negative.dm (hunk 2 above)
Comment (Kylerace) [reply code/datums/quirks/negative.dm:891]: the things i do for performance :trollface:

**Comment by mathis650** [comment]
based

**Comment by ExcessiveUseOfCobblestone** [comment]
I’m closing this if I see another troll face

**Comment by Kylerace** [comment]
alternatively you could merge it since its past 24 hours 😤

**Review by Timberpoes — APPROVED** [review]
:trollface:

**Comment by Kylerace** [comment]
:trollface:

**Pull request merged**